Src/WindowsServer/WindowsServer.Shared/AzureInstanceMetadataTelemetryModule.cs

```diff
@@ -40,10 +40,17 @@ public void Initialize(TelemetryConfiguration unused)
                                 // start off the heartbeat property collection process, but don't wait for it nor report
                                 // any status from here, fire and forget. The thread running the collection will report 
                                 // to the core event log.
-                                var heartbeatProperties = new AzureComputeMetadataHeartbeatPropertyProvider();
-                                Task.Factory.StartNew(
-                                    async () => await heartbeatProperties.SetDefaultPayloadAsync(hbeatManager)
-                                    .ConfigureAwait(false));
+                                try
+                                {
+                                    var heartbeatProperties = new AzureComputeMetadataHeartbeatPropertyProvider();
+                                    Task.Factory.StartNew(
+                                        async () => await heartbeatProperties.SetDefaultPayloadAsync(hbeatManager)
+                                        .ConfigureAwait(false));
+                                }
+                                catch (Exception heartbeatAquisitionException)
```

Comment:
This will not work. Because the task returned from StartNew is not awaited (it's fire-and-forget), exceptions won't be surfaced here. If the code in SetDefaultPayloadAsync throws an exception, the task will be Faulted and, since there's no-one to observe it, when the task is eventually finalized, you'll get an UnobservedTaskException event raised in the scheduler.

Instead, you need to add the try/catch inside SetDefaultPayloadAsync.

Reply:
I think perhaps catching them in both places may be prudent?